src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2675,6 +2675,15 @@ def query(self, filters=None, options=None):
                 ['id', '!^', f'{sys_config["basename"]}/'],
             ])
 
+            # top level dataset that stores all things related to gluster config
+            # needs to be hidden from local webUI since TrueCommander is
+            # responsible for configuring/managing gluster related settings
+            gluster_dataset = sys_config['basename'].split('/')[0] + '/.glusterfs'
```

Comment:
We could use `pool` instead of `basename`?

Reply:
Now that you bring it up, this wasn't proper anyways since the `.glusterfs` top-level dataset always exist on the zpool. Anyways, I've simplified the logic.